### PR TITLE
Upgrade to Fullscreen Edition

### PR DESCRIPTION
**Pull request by SirRichardFrancis**
## About The Pull Request

Firstly, this PR replaces build-in BYOND stat panel with browser-based TG stat panel (not the latest one, initial design looks so much better).

<details>
<summary>
Screenshots:
</summary>

image

image

image

image

image

image

image

image

</details>

As a consequence, the way verbs are added to client is now different, including admin verbs. That means little to the end user, 
<details>
<summary>
but fixed whatever the fuck that was:
</summary>

image

image

It's a list of lists of supposedly admin verbs that is supposedly sorted by permission type and is a reason why de-admin function came with massive lag and botched verbs for admeme that used it.
</details>

As another consequence, tabs "examine", "hardsuit", and "preference" were not supported by new stat panel and thus gone.
However, functionality formerly provided by "examine" tab is now all in the chat, courtesy of SPCR/@MLGTASTICa, who's "cool examines" I'm porting here.

What "hardsuit" tab had now will reside in "status" tab instead, with pictures and all.

Meanwhile "preference" tab will stay gone, for such settings are not something that really warrants a quick access.
That tab will make a comeback if it must, but that'll come with an extra delay.

Secondly, fullscreen mode is added along with changes to UI that make it prettier overall.

IMG_20240216_191755

Most notably, that one line at the bottom is gone and replaced with TG-style tooltip.

Before

image

After

image

Fullscreen mode is optional, but enabled by default. That could be changed in preferences or via F11 hotkey.

Thirdly, hotkey mode is gone, but "TAB" will change focus to chat input and back just like before.

## Why It's Good For The Game

Main advantage of new stat panel is not being laggy as hell. Ever noticed that switching to "Examine" tab gives noticeable performance improvement, but switching to "MC" tab turns on a slideshow? No more, performance is great at all times with this.*
*not mitigating lag caused by other things, obviously.

Fullscreen and UI changes here to make game look better.

Hotkey mode had been useless and I didn't see the point in even adding a button for it.

## Testing

A fair amount of touching things and making sure that they work correctly, but with literally every examine proc and verb affected, there is not way for every last thing to be tested fully by me alone.

## Changelog
:cl:
add: Fullscreen mode, hotkey is F11.
add: Browser-based stat panel from TG.
del: Extra lags from stat panel.
del: "Examine", "hardsuit", and "preference" tabs.
tweak: Changed how UI looks overall.
fix: Fixed de-admin functionality.
/:cl:

### CODE REVIEW

**Comment by thevandie** [comment]
👍 👍 👍 👍 👍 👍 
I love tg QOL I love tg QOL I love tg QOL I love tg QOL

**Comment by ValoTheValo** [comment]
finally, after 10000 years i can be able to see whats going on in game

**Comment by SirRichardFrancis** [comment]
Ready for review now.
Description and screenshots updated.

**Review by Humonitarian — APPROVED** [review]
I saw nothing wrong at first glance

**Comment by SirRichardFrancis** [comment]
> Seems to have a few issues on server that I've noticed: Storyteller not appearing in status tab (at least for me, not sure how widespread that is) Needing to relog to load new tabs (i.e. if I spawn in as a borer, my ghost tab stays and my abilities tab doesn't appear until I relog)

Noted. Though you don't have to relog, there is a 'Fix Stat Panel' verb in OOC tab, for cases when verbs and tabs didn't update for some reason (like here with spawning as a borer).

> Not sure WHY these are happening, as it seems to work when you tested it locally.

Testing literally every possible situation involving verbs would have taken me weeks of incredibly boring solo autism gameplay. Didn't exactly do that and warned about it.

Not sure WHY this is a question, as it seems to have been in PR description when you read it.

**Comment by SirRichardFrancis** [comment]
List of issues reported during live test with fixes checklist:

- [x] NT cruciform energy display is broken.
- [ ] Carrion hunger and chemical storage displays are broken.
- [ ] Cybernetic, borer and carrion tabs do not appear when they should.
- [ ] Turf content icons are sometimes missing, 'bad resource file' runtime.
- [ ] Whisper and several other verbs could be duplicated sometimes
- [ ] Storyteller isn't in the status tab
- [ ] Status tab lists players and their job of choice in a way that isn't pretty

**Comment by TheShown911** [comment]
Uh oh, merge conflict...

**Comment by SirRichardFrancis** [comment]
final_fix

**Comment by SirRichardFrancis** [comment]
Another Friday, another merge conflict...

File: code/modules/mechs/mech.dm [hunk 1]
```diff
@@ -193,50 +193,47 @@
 /mob/living/exosuit/IsAdvancedToolUser()
 	return TRUE
 
-/mob/living/exosuit/examine(mob/user)
-	. = ..()
-	if(LAZYLEN(pilots) && (!hatch_closed || body.pilot_coverage < 100 || body.transparent_cabin || (body && !body.has_hatch)))
-		to_chat(user, "It is being piloted by [english_list(pilots, nothing_text = "nobody")].")
+/mob/living/exosuit/examine(mob/user, extra_description = "")
+	if(LAZYLEN(pilots) && (!hatch_closed || body.pilot_coverage < 100 || body.transparent_cabin || (body && !body.has_hatch))
```
Comment (SirRichardFrancis) [inline code/modules/mechs/mech.dm:197]:
```suggestion
	if(LAZYLEN(pilots) && (!hatch_closed || body.pilot_coverage < 100 || body.transparent_cabin || (body && !body.has_hatch)))
```

**Comment by SirRichardFrancis** [comment]
Bruh. Should have just done it on local.

**Pull request merged**